processing/test_visualise.cpp

```diff
@@ -288,26 +288,58 @@ void printVehicleInfo(int objectID, std::map<int, std::vector<BoundingBox>> vehi
   std::cout << "==================" << std::endl;
 }
 
+void dumpVehicleInfo(std::vector<BoundingBox> vehicleInfo) {
+  std::cout << "==START VEHICLE INFO DUMP==" << std::endl;
+  int i = 0;
+  for (auto it = vehicleInfo.begin(); it != vehicleInfo.end(); ++it) {
+    std::cout << "==================" << std::endl;
+    std::cout << "Frame [" << i << "]:" << std::endl;
+    std::cout << "  Centroid: " << it->centroid.x << " " << it->centroid.y << " " << it->centroid.z << std::endl;
+    std::cout << "==================" << std::endl;
+    i++;
+  }
+  std::cout << "==END DUMP VEHICLE INFO DUMP==" << std::endl;
+}
+
+void dumpAll(std::map<int, std::vector<BoundingBox>> &all) {
+  std::cout << "=======>START DUMP ALL<===========" << std::endl;
+  for (auto map_it = all.begin(); map_it != all.end(); ++map_it) {
+    std::cout << "VEHICLE ID: [" << map_it->first << "]" << std::endl;
+    dumpVehicleInfo(map_it->second);
+  }
+  std::cout << "=======>END DUMP ALL<===========" << std::endl;
+}
+
 
 // Object tracker derived from: https://www.pyimagesearch.com/2018/07/23/simple-object-tracking-with-opencv/
 void register_new (pcl::PointXYZ centroid, std::map<int, pcl::PointXYZ> &objects, std::map<int, int> &disappeared,  int &nextObjectID ) {
   objects[nextObjectID] = centroid;
   disappeared[nextObjectID] = 0;
+  // std::cout << "POST REGISTER:" << std::endl;
+  // std::cout << "Num disappeared:" << disappeared.size() << std::endl;
+  // std::cout << "Disappeared: ";
+  // for (auto it = disappeared.begin(); it != disappeared.end(); ++it){
+  //   std::cout << "(" << it->first << "," << it->second << ") ";
+  // }
+  // std::cout << std::endl;
   nextObjectID++;
 }
 
 void update(std::vector<pcl::PointXYZ> inputCentroids, std::map<int, pcl::PointXYZ> &objects, std::map<int, int> &disappeared, int &nextObjectID, int& totalCount, std::map<int, std::vector<BoundingBox>>& vehicleInfo ) {
-
+  //dumpAll(vehicleInfo);
   if (inputCentroids.size() == 0) {
     for (auto it = disappeared.begin(); it != disappeared.end(); ++it) {
       it->second++;
     }
     for (auto it = disappeared.cbegin(); it != disappeared.cend(); ) {
       if (it->second > MAX_FRAMES_DISAPPEARED) {
+        //std::cout << "VEHICLE [" << it->first << "] EXCEEDED DISAPPEAR #1 THRESHOLD" << std::endl;
         printVehicleInfo(it->first, vehicleInfo);
+        //dumpVehicleInfo(vehicleInfo[it->first]);
         objects.erase(it->first);
-        disappeared.erase(it++);
```

Comment:
iterator was incorrectly advanced before it->first had been read to delete the correct information from vehicleInfo
this was causing some vehicles to apparently have appeared for 0 frames and be classified as a motorcycle